eZ/Publish/Core/Repository/RoleService.php

```diff
@@ -181,6 +183,58 @@ public function createRoleDraft(APIRole $role): APIRoleDraft
         return $this->roleDomainMapper->buildDomainRoleDraftObject($spiRole);
     }
 
+    public function copyRole(APIRole $role, APIRoleCopyStruct $roleCopyStruct): APIRole
+    {
+        if (!is_string($roleCopyStruct->newIdentifier) || empty($roleCopyStruct->newIdentifier)) {
+            throw new InvalidArgumentValue('newIdentifier', $roleCopyStruct->newIdentifier, 'RoleCopyStruct');
+        }
+
+        if (!$this->permissionResolver->canUser('role', 'create', $roleCopyStruct)) {
+            throw new UnauthorizedException('role', 'create');
+        }
+
+        try {
+            $existingRole = $this->loadRoleByIdentifier($roleCopyStruct->newIdentifier);
+
+            throw new InvalidArgumentException(
+                '$roleCopyStruct',
+                "Role '{$existingRole->id}' with the specified identifier '{$roleCopyStruct->newIdentifier}' " .
+                'already exists'
+            );
+        } catch (APINotFoundException $e) {
+            // Do nothing
+        }
+
+        foreach ($role->getPolicies() as $policy) {
+            $policyCreateStruct = new PolicyCreateStruct([
+                'module' => $policy->module,
+                'function' => $policy->function,
+            ]);
+            foreach ($policy->getLimitations() as $limitation) {
+                $policyCreateStruct->addLimitation($limitation);
+            }
+            $roleCopyStruct->addPolicy($policyCreateStruct);
+        }
+
+        $limitationValidationErrors = $this->validateRoleCopyStruct($roleCopyStruct);
+        if (!empty($limitationValidationErrors)) {
+            throw new LimitationValidationException($limitationValidationErrors);
+        }
+
+        $spiRoleCopyStruct = $this->roleDomainMapper->buildPersistenceRoleCopyStruct($roleCopyStruct, $role->id);
```

Comment:
CS, line is too long and it doesn't fit on a screen:
```suggestion
        $spiRoleCopyStruct = $this->roleDomainMapper->buildPersistenceRoleCopyStruct(
            $roleCopyStruct,
            $role->id
        );
```